volatility3/framework/layers/crash.py

```diff
@@ -67,34 +72,96 @@ def __init__(self, context: interfaces.context.ContextInterface, config_path: st
             vollog.log(constants.LOGLEVEL_VVVV, "unsupported dump format 0x{:x}".format(header.DumpType))
             raise WindowsCrashDumpFormatException(name, "unsupported dump format 0x{:x}".format(header.DumpType))
 
+        # Then call the super, which will call load_segments (which needs the base_layer before it'll work)
         super().__init__(context, config_path, name)
 
+    def get_header(self) -> interfaces.objects.ObjectInterface:
+        return self.context.object(self._crash_table_name + constants.BANG + self.dump_header_name,
+                                   offset=0,
+                                   layer_name=self._base_layer)
+
+    def get_summary_header(self) -> interfaces.objects.ObjectInterface:
+        return self.context.object(self._crash_common_table_name + constants.BANG + "_SUMMARY_DUMP",
+                                   offset=0x1000 * self.headerpages,
+                                   layer_name=self._base_layer)
+
     def _load_segments(self) -> None:
         """Loads up the segments from the meta_layer."""
-        header = self.context.object(self._crash_table_name + constants.BANG + self.dump_header_name,
-                                     offset = 0,
-                                     layer_name = self._base_layer)
 
         segments = []
 
-        offset = self.headerpages
-        header.PhysicalMemoryBlockBuffer.Run.count = header.PhysicalMemoryBlockBuffer.NumberOfRuns
-        for x in header.PhysicalMemoryBlockBuffer.Run:
-            segments.append((x.BasePage * 0x1000, offset * 0x1000, x.PageCount * 0x1000, x.PageCount * 0x1000))
-            # print("Segments {:x} {:x} {:x}".format(x.BasePage * 0x1000,
-            #                  offset * 0x1000,
-            #                  x.PageCount * 0x1000))
-            offset += x.PageCount
+        if self.dump_type == 0x1:
+            header = self.context.object(self._crash_table_name + constants.BANG + self.dump_header_name,
+                                         offset=0,
+                                         layer_name=self._base_layer)
+
+            offset = self.headerpages
+            header.PhysicalMemoryBlockBuffer.Run.count = header.PhysicalMemoryBlockBuffer.NumberOfRuns
+            for x in header.PhysicalMemoryBlockBuffer.Run:
```

Comment:
Not a deal breaker by any means, but may be nice to have a more descriptive variable name than just `x`?  I know its further use it only two lines aware, but even just `run` would be better?

Reply:
Fixed in a0adbde